File: devops-scm-spring-boot-starter/src/main/java/com/tencent/devops/scm/spring/properties/HttpClientProperties.java
package com.tencent.devops.scm.spring.properties;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class HttpClientProperties {

    /**
     * api url
     */
    private String apiUrl;

    // 连接超时,单位: SECONDS
    @Builder.Default
    private Long connectTimeout = 5L;
    // 读取超时,单位: SECONDS
    @Builder.Default
    private Long readTimeout = 30L;
    // 写超时,单位: SECONDS
    @Builder.Default
    private Long writeTimeout = 30L;
    // 启用度量
    @Builder.Default
    private Boolean metrics = false;
}

File: devops-scm-spring-boot-starter/src/main/kotlin/com/tencent/devops/scm/spring/properties/HttpClientProperties.kt
package com.tencent.devops.scm.spring.properties

/**
 * HTTP客户端配置属性
 *
 * @property apiUrl API地址
 * @property connectTimeout 连接超时时间（单位：秒），默认5秒
 * @property readTimeout 读取超时时间（单位：秒），默认30秒
 * @property writeTimeout 写入超时时间（单位：秒），默认30秒
 * @property metrics 是否启用度量，默认false
 */
data class HttpClientProperties(
    var apiUrl: String? = null,
    var connectTimeout: Long = 5L,
    var readTimeout: Long = 30L,
    var writeTimeout: Long = 30L,
    var metrics: Boolean = false
)